lib/index.js

```diff
@@ -34,12 +34,13 @@ function response (httpResponse, data) {
     },
     json () {
       return new Promise((resolve, reject) => {
-        const err = MOPointer.alloc().initWithValue(null)
-        const obj = NSJSONSerialization.JSONObjectWithData_options_error(data, 0, err)
-        if (obj) {
+        const str = NSString.alloc().initWithData_encoding(data, NSUTF8StringEncoding)
+        if (str) {
+          // parse errors are turned into exceptions, which cause promise to be rejected
+          const obj = JSON.parse(str)
           resolve(obj)
         } else {
-          reject(err.value())
+          reject(new Error("Could not parse JSON because it is not valid UTF-8 data."))
```

Comment:
shouldn't we throw the parsing error as well?
`JSON.parse` will throw if the string is not a JSON object but it could still be a "valid UTF-8 data" (like `undefined`).